packages/core/src/components/spinner/spinner.tsx

```diff
@@ -76,81 +76,21 @@ export interface SpinnerProps<T extends HTMLElement = HTMLElement> extends Props
  *
  * @see https://blueprintjs.com/docs/#core/components/spinner
  */
-export class Spinner extends AbstractPureComponent<SpinnerProps> {
-    public static displayName = `${DISPLAYNAME_PREFIX}.Spinner`;
+export const Spinner: React.FC<SpinnerProps> = props => {
+    const { className = "", intent, value, tagName = "div", size, ...htmlProps } = props;
 
-    public componentDidUpdate(prevProps: SpinnerProps) {
-        if (prevProps.value !== this.props.value) {
-            // IE/Edge: re-render after changing value to force SVG update
-            this.forceUpdate();
-        }
-    }
-
-    public render() {
-        const { className, intent, value, tagName = "div", ...htmlProps } = this.props;
-        const size = this.getSize();
-
-        const classes = classNames(
-            Classes.SPINNER,
-            Classes.intentClass(intent),
-            { [Classes.SPINNER_NO_SPIN]: value != null },
-            className,
-        );
-
-        // keep spinner track width consistent at all sizes (down to about 10px).
-        const strokeWidth = Math.min(MIN_STROKE_WIDTH, (STROKE_WIDTH * SpinnerSize.LARGE) / size);
-        const strokeOffset = PATH_LENGTH - PATH_LENGTH * (value == null ? 0.25 : clamp(value, 0, 1));
-
-        // multiple DOM elements around SVG are necessary to properly isolate animation:
-        // - SVG elements in IE do not support anim/trans so they must be set on a parent HTML element.
-        // - SPINNER_ANIMATION isolates svg from parent display and is always centered inside root element.
-        return React.createElement(
-            tagName,
-            {
-                "aria-label": "loading",
-                "aria-valuemax": 100,
-                "aria-valuemin": 0,
-                "aria-valuenow": value === undefined ? undefined : value * 100,
-                className: classes,
-                role: "progressbar",
-                ...htmlProps,
-            },
-            React.createElement(
-                tagName,
-                { className: Classes.SPINNER_ANIMATION },
-                <svg
-                    width={size}
-                    height={size}
-                    strokeWidth={strokeWidth.toFixed(2)}
-                    viewBox={this.getViewBox(strokeWidth)}
-                >
-                    <path className={Classes.SPINNER_TRACK} d={SPINNER_TRACK} />
-                    <path
-                        className={Classes.SPINNER_HEAD}
-                        d={SPINNER_TRACK}
-                        pathLength={PATH_LENGTH}
-                        strokeDasharray={`${PATH_LENGTH} ${PATH_LENGTH}`}
-                        strokeDashoffset={strokeOffset}
-                    />
-                </svg>,
-            ),
-        );
-    }
-
-    protected validateProps({ className = "", size }: SpinnerProps) {
+    React.useEffect(() => {
```

Comment:
This was happening in `validateProps` before which is only called if the node env is not production: https://github.com/palantir/blueprint/blob/f374289ad883ff00f71b84cc77180183f0166a5e/packages/core/src/common/abstractComponent.ts#L53-L55

We should make sure we're not doing this check in production. Maybe we can add a `useValidateProps` hooks that only calls the callback if the env is not production?

Reply:
Good catch! Will add this 👍

Reply:
Extracted to a hook here: b36f43a819e965447d596173bbbeb0cca2686f6b